examples/3-adapter_advanced.ipynb

```diff
@@ -0,0 +1,740 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step = Step(name='xgboost',\n",
+    "               transformer=XGBoostTransformer(xgb_params=get_xgb_params(), num_boost_round=5),\n",
+    "               input_steps=[rename_step],\n",
+    "               cache_dirpath=CACHE_DIR,\n",
+    "               force_fitting=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xgb_step.fit_transform(data_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All right, so now we have two models trained, but we haven't checked so far how do they perform. Let's do it now. We will use one of data scientists' favourite measures: log-loss."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rf_proba_train = rf_step.transform(data_train)['y_proba']\n",
+    "rf_proba_test = rf_step.transform(data_test)['y_proba']\n",
+    "xgb_proba_train = xgb_step.transform(data_train)['y_proba']\n",
+    "xgb_proba_test = xgb_step.transform(data_test)['y_proba']\n",
+    "\n",
+    "print(\"RF train: {:.3f}, test: {:.3f}\".format(log_loss(y_pred=rf_proba_train, y_true=y_train),\n",
+    "                                      log_loss(y_pred=rf_proba_test, y_true=y_test)))\n",
+    "print(\"XGB train: {:.3f}, test: {:.3f}\".format(log_loss(y_pred=xgb_proba_train, y_true=y_train),\n",
+    "                                       log_loss(y_pred=xgb_proba_test, y_true=y_test)))\n",
+    "print(\"Averaged predictions: {:.3f}, test: {:.3f}\".format(log_loss(y_pred=(rf_proba_train + xgb_proba_train) / 2, y_true=y_train),\n",
+    "                                            log_loss(y_pred=(rf_proba_test + xgb_proba_test) / 2, y_true=y_test)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that we noticeably overfit, but it's OK - making a perfect model is not the goal of this notebook. We also see that we could benefit from a very simple model ensembling: averaging of model predictions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Ensembling and evaluation steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next step incorporates averaging of model predictions into the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ensemble_step = Step(name='ensembler',\n",
+    "                 transformer=Dummy(),\n",
+    "                 input_steps=[rf_step, xgb_step],                 \n",
+    "                 adapter={'y_proba': ([(rf_step.name, 'y_proba'),\n",
```

Comment:
So we made the adapter do all the computation and the transformer does nothing? This seems counter-intuitive to me...